pandas/core/indexes/base.py

```diff
@@ -3981,8 +3981,11 @@ def _should_partial_index(self, target: Index) -> bool:
         Should we attempt partial-matching indexing?
         """
         if is_interval_dtype(self.dtype):
+            if is_interval_dtype(target.dtype):
+                return False
             # "Index" has no attribute "left"
-            return self.left._should_compare(target)  # type: ignore[attr-defined]
+            # return self.left._should_compare(target)  # type: ignore[attr-defined]
```

Comment:
Should this commented out line work in the future or is it leftover from investigating this fix?

Reply:
Yes, I think so that ideally it gets restored in the future. I added a comment referencing the issue I had opened about that